File: lib/features/onboarding/presentation/onboarding_screen.dart
// ============================================================
// FILE: lib/features/onboarding/presentation/onboarding_screen.dart
// PURPOSE: First-launch onboarding flow — permission setup,
//          goal selection, wallpaper picker, accent color.
// CREATED: 2026-08-03 | LAST MODIFIED: 2026-08-03
// ============================================================
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import 'package:go_router/go_router.dart';
import 'package:image_picker/image_picker.dart';
import '../../../core/theme/glass_tokens.dart';
import '../../../shared/widgets/glass_card.dart';
import '../../../shared/widgets/glass_button.dart';
import '../../../core/native_bridge/accessibility_bridge.dart';
import '../../../core/native_bridge/usage_stats_bridge.dart';
import '../../../core/native_bridge/overlay_bridge.dart';
import '../../../core/theme/wallpaper_controller.dart';
import '../../../core/state/onboarding_controller.dart';

class OnboardingScreen extends ConsumerStatefulWidget {
  const OnboardingScreen({super.key});

  @override
  ConsumerState<OnboardingScreen> createState() => _OnboardingScreenState();
}

class _OnboardingScreenState extends ConsumerState<OnboardingScreen> {
  final PageController _pageController = PageController();
  int _currentPage = 0;

  final _goals = [
    _GoalOption('Study & Exam Prep', Icons.school_outlined, 'Block distractions during study sessions'),
    _GoalOption('General Focus', Icons.center_focus_strong, 'Build better focus habits'),
    _GoalOption('Digital Detox', Icons.phone_android_outlined, 'Reduce screen time significantly'),
  ];

  String _selectedGoal = '';
  Color _selectedAccent = GlassTokens.accentPrimary;
  final _accentColors = [
    const Color(0xFF7C5CFF), // Violet
    const Color(0xFF60A5FA), // Blue
    const Color(0xFF3DDC97), // Green
    const Color(0xFFFFC857), // Amber
    const Color(0xFFFF5C7C), // Rose
    const Color(0xFF9B7FFF), // Lavender
  ];

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: GlassTokens.bgBase,
      body: SafeArea(
        child: Column(
          children: [
            // Page indicator
            Padding(
              padding: const EdgeInsets.symmetric(vertical: 16),
              child: Row(
                mainAxisAlignment: MainAxisAlignment.center,
                children: List.generate(5, (i) => _buildIndicator(i)),
              ),
            ),
            // Page view
            Expanded(
              child: PageView(
                controller: _pageController,
                onPageChanged: (page) => setState(() => _currentPage = page),
                physics: const NeverScrollableScrollPhysics(),
                children: [
                  _buildWelcomePage(),
                  _buildPermissionsPage(),
                  _buildGoalPage(),
                  _buildWallpaperPage(),
                  _buildThemePage(),
                ],
              ),
            ),
            // Navigation
            Padding(
              padding: const EdgeInsets.all(24),
              child: Row(
                children: [
                  if (_currentPage > 0)
                    Expanded(
                      child: GlassButton(
                        label: 'Back',
                        isPrimary: false,
                        onPressed: () {
                          _pageController.previousPage(
                            duration: const Duration(milliseconds: 300),
                            curve: Curves.easeInOut,
                          );
                        },
                      ),
                    ),
                  if (_currentPage > 0) const SizedBox(width: 12),
                  Expanded(
                    flex: 2,
                    child: GlassButton(
                      label: _currentPage == 4 ? 'Get Started' : 'Next',
                      icon: _currentPage == 4
                          ? Icons.check
                          : Icons.arrow_forward,
                      isPrimary: true,
                      onPressed: () async {
                        if (_currentPage == 4) {
                          await ref
                              .read(onboardingCompletedProvider.notifier)
                              .markCompleted();
                          if (context.mounted) context.go('/');
                        } else {
                          _pageController.nextPage(
                            duration: const Duration(milliseconds: 300),
                            curve: Curves.easeInOut,
                          );
                        }
                      },
                    ),
                  ),
                ],
              ),
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildIndicator(int index) {
    final isActive = index == _currentPage;
    return AnimatedContainer(
      duration: const Duration(milliseconds: 300),
      margin: const EdgeInsets.symmetric(horizontal: 4),
      width: isActive ? 24 : 8,
      height: 8,
      decoration: BoxDecoration(
        color: isActive
            ? GlassTokens.accentPrimary
            : Colors.white.withValues(alpha: 0.2),
        borderRadius: BorderRadius.circular(4),
      ),
    );
  }

  Widget _buildWelcomePage() {
    return Padding(
      padding: const EdgeInsets.all(32),
      child: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          Icon(
            Icons.shield_outlined,
            size: 80,
            color: GlassTokens.accentPrimary,
          ),
          const SizedBox(height: 32),
          Text(
            'Welcome to FocusForge',
            style: TextStyle(
              color: GlassTokens.textPrimary,
              fontSize: 28,
              fontWeight: FontWeight.bold,
            ),
            textAlign: TextAlign.center,
          ),
          const SizedBox(height: 16),
          Text(
            'Your focus companion with liquid glass design.\nBlock distractions. Study smarter. Stay in flow.',
            style: TextStyle(
              color: GlassTokens.textSecondary,
              fontSize: 16,
              height: 1.5,
            ),
            textAlign: TextAlign.center,
          ),
        ],
      ),
    );
  }

  Widget _buildPermissionsPage() {
    return Padding(
      padding: const EdgeInsets.all(24),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Text(
            'Set Up Permissions',
            style: TextStyle(
              color: GlassTokens.textPrimary,
              fontSize: 24,
              fontWeight: FontWeight.bold,
            ),
          ),
          const SizedBox(height: 8),
          Text(
            'FocusForge needs these to protect your focus.',
            style: TextStyle(color: GlassTokens.textSecondary),
          ),
          const SizedBox(height: 24),
          _PermissionTile(
            icon: Icons.accessibility_new,
            title: 'Accessibility Service',
            subtitle: 'Detects which app is in the foreground',
            onTap: () => ref.read(accessibilityBridgeProvider).openSettings(),
          ),
          _PermissionTile(
            icon: Icons.history,
            title: 'Usage Access',
            subtitle: 'Tracks screen time for your dashboard',
            onTap: () => ref.read(usageStatsBridgeProvider).openSettings(),
          ),
          _PermissionTile(
            icon: Icons.layers_outlined,
            title: 'Draw Over Apps',
            subtitle: 'Shows block overlay screens',
            onTap: () => ref.read(overlayBridgeProvider).openSettings(),
          ),
        ],
      ),
    );
  }

  Widget _buildGoalPage() {
    return Padding(
      padding: const EdgeInsets.all(24),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Text(
            'What\'s your goal?',
            style: TextStyle(
              color: GlassTokens.textPrimary,
              fontSize: 24,
              fontWeight: FontWeight.bold,
            ),
          ),
          const SizedBox(height: 24),
          ...(_goals.map((goal) => _buildGoalCard(goal))),
        ],
      ),
    );
  }

  Widget _buildGoalCard(_GoalOption goal) {
    final isSelected = _selectedGoal == goal.title;
    return Padding(
      padding: const EdgeInsets.only(bottom: 12),
      child: GlassCard(
        padding: const EdgeInsets.all(16),
        onTap: () => setState(() => _selectedGoal = goal.title),
        child: Row(
          children: [
            Container(
              width: 48,
              height: 48,
              decoration: BoxDecoration(
                color: isSelected
                    ? GlassTokens.accentPrimary.withValues(alpha: 0.2)
                    : Colors.white.withValues(alpha: 0.06),
                borderRadius: BorderRadius.circular(12),
              ),
              child: Icon(
                goal.icon,
                color: isSelected
                    ? GlassTokens.accentPrimary
                    : GlassTokens.textSecondary,
              ),
            ),
            const SizedBox(width: 16),
            Expanded(
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Text(
                    goal.title,
                    style: TextStyle(
                      color: GlassTokens.textPrimary,
                      fontSize: 16,
                      fontWeight: FontWeight.w600,
                    ),
                  ),
                  Text(
                    goal.subtitle,
                    style: TextStyle(
                      color: GlassTokens.textSecondary,
                      fontSize: 13,
                    ),
                  ),
                ],
              ),
            ),
            if (isSelected)
              Icon(Icons.check_circle, color: GlassTokens.accentPrimary),
          ],
        ),
      ),
    );
  }

  Widget _buildWallpaperPage() {
    return Padding(
      padding: const EdgeInsets.all(24),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Text(
            'Choose Your Wallpaper',
            style: TextStyle(
              color: GlassTokens.textPrimary,
              fontSize: 24,
              fontWeight: FontWeight.bold,
            ),
          ),
          const SizedBox(height: 8),
          Text(
            'Your wallpaper shows behind the glass panels.',
            style: TextStyle(color: GlassTokens.textSecondary),
          ),
          const SizedBox(height: 24),
          GlassCard(
            onTap: () async {
              final picker = ImagePicker();
              final image = await picker.pickImage(source: ImageSource.gallery);
              if (image != null) {
                ref.read(wallpaperProvider.notifier).setWallpaper(image.path);
              }
            },
            child: Column(
              children: [
                Icon(
                  Icons.photo_library_outlined,
                  size: 48,
                  color: GlassTokens.accentPrimary,
                ),
                const SizedBox(height: 12),
                Text(
                  'Pick from Gallery',
                  style: TextStyle(
                    color: GlassTokens.textPrimary,
                    fontSize: 16,
                    fontWeight: FontWeight.w500,
                  ),
                ),
                const SizedBox(height: 4),
                Text(
                  'Or use the default glass background',
                  style: TextStyle(
                    color: GlassTokens.textSecondary,
                    fontSize: 13,
                  ),
                ),
              ],
            ),
          ),
        ],
      ),
    );
  }

  Widget _buildThemePage() {
    return Padding(
      padding: const EdgeInsets.all(24),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Text(
            'Pick Accent Color',
            style: TextStyle(
              color: GlassTokens.textPrimary,
              fontSize: 24,
              fontWeight: FontWeight.bold,
            ),
          ),
          const SizedBox(height: 24),
          Wrap(
            spacing: 16,
            runSpacing: 16,
            children: _accentColors.map((color) {
              final isSelected = _selectedAccent == color;
              return GestureDetector(
                onTap: () => setState(() {
                  _selectedAccent = color;
                  GlassTokens.accentPrimary = color;
                }),
                child: Container(
                  width: 56,
                  height: 56,
                  decoration: BoxDecoration(
                    color: color,
                    shape: BoxShape.circle,
                    border: Border.all(
                      color: isSelected ? Colors.white : Colors.transparent,
                      width: 3,
                    ),
                    boxShadow: isSelected
                        ? [
                            BoxShadow(
                              color: color.withValues(alpha: 0.5),
                              blurRadius: 16,
                              spreadRadius: 2,
                            )
                          ]
                        : null,
                  ),
                  child: isSelected
                      ? const Icon(Icons.check, color: Colors.white)
                      : null,
                ),
              );
            }).toList(),
          ),
          const SizedBox(height: 32),
          GlassCard(
            child: Text(
              'Preview: The glass panels will use this accent color for buttons, progress rings, and active states.',
              style: TextStyle(
                color: GlassTokens.textSecondary,
                fontSize: 14,
              ),
            ),
          ),
        ],
      ),
    );
  }
}

class _GoalOption {
  final String title;
  final IconData icon;
  final String subtitle;

  _GoalOption(this.title, this.icon, this.subtitle);
}

class _PermissionTile extends StatelessWidget {
  final IconData icon;
  final String title;
  final String subtitle;
  final VoidCallback onTap;

  const _PermissionTile({
    required this.icon,
    required this.title,
    required this.subtitle,
    required this.onTap,
  });

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.only(bottom: 12),
      child: GlassCard(
        padding: const EdgeInsets.all(16),
        onTap: onTap,
        child: Row(
          children: [
            Icon(icon, color: GlassTokens.accentPrimary, size: 28),
            const SizedBox(width: 16),
            Expanded(
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Text(
                    title,
                    style: TextStyle(
                      color: GlassTokens.textPrimary,
                      fontSize: 15,
                      fontWeight: FontWeight.w600,
                    ),
                  ),
                  Text(
                    subtitle,
                    style: TextStyle(
                      color: GlassTokens.textSecondary,
                      fontSize: 12,
                    ),
                  ),
                ],
              ),
            ),
            Icon(Icons.chevron_right, color: GlassTokens.textSecondary),
          ],
        ),
      ),
    );
  }
}

File: app/src/main/java/com/example/ui/screens/OnboardingScreen.kt
// ============================================================
// FILE: app/src/main/java/com/example/ui/screens/OnboardingScreen.kt
// PURPOSE: First-launch onboarding flow with permission setup (Accessibility,
//          Usage Stats, Overlay, Notifications) and goal picker.
// CREATED: 2026-08-09
// ============================================================

package com.example.ui.screens

import android.content.Intent
import android.provider.Settings
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.ui.theme.*

@Composable
fun OnboardingScreen(
    onFinishOnboarding: () -> Unit
) {
    val context = LocalContext.current
    var step by remember { mutableIntStateOf(1) }

    WallpaperBackground(preset = "COSMIC_NEON") {
        Box(
            modifier = Modifier.fillMaxSize().padding(24.dp),
            contentAlignment = Alignment.Center
        ) {
            when (step) {
                1 -> {
                    GlassCard(modifier = Modifier.fillMaxWidth()) {
                        Column(horizontalAlignment = Alignment.CenterHorizontally) {
                            Icon(
                                imageVector = Icons.Default.Shield,
                                contentDescription = null,
                                tint = GlassTokens.Info,
                                modifier = Modifier.size(64.dp)
                            )
                            Spacer(modifier = Modifier.height(16.dp))
                            Text(
                                text = "Welcome to FocusForge",
                                fontSize = 26.sp,
                                fontWeight = FontWeight.Bold,
                                color = GlassTokens.TextPrimary,
                                textAlign = TextAlign.Center
                            )
                            Spacer(modifier = Modifier.height(8.dp))
                            Text(
                                text = "Reclaim your screen time with Liquid Glass dark design, Reels/Shorts Blocker, and Pomodoro study tools.",
                                fontSize = 14.sp,
                                color = GlassTokens.TextSecondary,
                                textAlign = TextAlign.Center
                            )
                            Spacer(modifier = Modifier.height(28.dp))
                            GlassButton(
                                text = "Get Started",
                                onClick = { step = 2 },
                                modifier = Modifier.fillMaxWidth(),
                                icon = Icons.Default.ArrowForward,
                                testTagStr = "onboarding_next_button"
                            )
                        }
                    }
                }
                2 -> {
                    GlassCard(modifier = Modifier.fillMaxWidth()) {
                        Column {
                            Text(
                                text = "Enable System Permissions",
                                fontSize = 22.sp,
                                fontWeight = FontWeight.Bold,
                                color = GlassTokens.TextPrimary
                            )
                            Spacer(modifier = Modifier.height(8.dp))
                            Text(
                                text = "FocusForge uses native Android Accessibility & Usage access to block distractions directly on your device.",
                                fontSize = 13.sp,
                                color = GlassTokens.TextSecondary
                            )
                            Spacer(modifier = Modifier.height(20.dp))

                            PermissionItem(
                                title = "Accessibility Service",
                                description = "Required to detect Instagram Reels, YouTube Shorts & blocked apps",
                                icon = Icons.Default.AccessibilityNew,
                                onClick = {
                                    context.startActivity(Intent(Settings.ACTION_ACCESSIBILITY_SETTINGS))
                                }
                            )

                            Spacer(modifier = Modifier.height(12.dp))

                            PermissionItem(
                                title = "Usage Stats Access",
                                description = "Required to generate screen time analytics & dashboards",
                                icon = Icons.Default.Analytics,
                                onClick = {
                                    context.startActivity(Intent(Settings.ACTION_USAGE_ACCESS_SETTINGS))
                                }
                            )

                            Spacer(modifier = Modifier.height(12.dp))

                            PermissionItem(
                                title = "Display Over Other Apps",
                                description = "Required to show liquid glass block overlays",
                                icon = Icons.Default.Layers,
                                onClick = {
                                    context.startActivity(Intent(Settings.ACTION_MANAGE_OVERLAY_PERMISSION))
                                }
                            )

                            Spacer(modifier = Modifier.height(24.dp))

                            GlassButton(
                                text = "Continue to App",
                                onClick = onFinishOnboarding,
                                modifier = Modifier.fillMaxWidth(),
                                testTagStr = "onboarding_finish_button"
                            )
                        }
                    }
                }
            }
        }
    }
}

@Composable
private fun PermissionItem(
    title: String,
    description: String,
    icon: androidx.compose.ui.graphics.vector.ImageVector,
    onClick: () -> Unit
) {
    Surface(
        modifier = Modifier.fillMaxWidth(),
        shape = RoundedCornerShape(16.dp),
        color = Color.White.copy(alpha = 0.08f)
    ) {
        Row(
            modifier = Modifier.padding(14.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            Icon(
                imageVector = icon,
                contentDescription = null,
                tint = GlassTokens.Accent,
                modifier = Modifier.size(28.dp)
            )
            Spacer(modifier = Modifier.width(12.dp))
            Column(modifier = Modifier.weight(1f)) {
                Text(
                    text = title,
                    fontSize = 15.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = GlassTokens.TextPrimary
                )
                Text(
                    text = description,
                    fontSize = 11.sp,
                    color = GlassTokens.TextSecondary
                )
            }
            Spacer(modifier = Modifier.width(8.dp))
            Button(
                onClick = onClick,
                colors = ButtonDefaults.buttonColors(containerColor = GlassTokens.Accent),
                shape = RoundedCornerShape(12.dp)
            ) {
                Text("Enable", fontSize = 12.sp)
            }
        }
    }
}
